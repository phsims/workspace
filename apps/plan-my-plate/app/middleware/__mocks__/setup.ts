export const getSpoonConfig = () => ({
  baseUrl: 'http://mock_base_url.com',
   imgUrl: 'http://mock_img_url.com',
  headers: {
    'Content-Type': 'application/json',
    'X-API-Key': 'http://my_mock_api_key.com',
  },
  env: 'development',
});
