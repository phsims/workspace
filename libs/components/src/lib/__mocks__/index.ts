export * from './nutrientsData'
export * from './recipesData'