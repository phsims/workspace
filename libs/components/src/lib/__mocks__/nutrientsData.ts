
export const mockNutrients= [
    {
        "name": "Calories",
        "amount": 316.49,
        "unit": "kcal",
        "percentOfDailyNeeds": 15.82
    },
    {
        "name": "Fat",
        "amount": 12.09,
        "unit": "g",
        "percentOfDailyNeeds": 18.6
    },
]