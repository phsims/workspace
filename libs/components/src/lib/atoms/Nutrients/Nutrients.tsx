import { Box, Typography } from '@mui/material';

export interface NutrientsProps {
  nutrients: Nutrient[];
  title?: string;
}

export interface Nutrient {
  name: string;
  amount: number;
  unit: string;
}

export function Nutrients({ nutrients, title }: NutrientsProps) {
  const viewNutrients = ['Calories', 'Fat', 'Carbohydrates', 'Protein'];
  const filteredNutrients = nutrients
    ? nutrients.filter((nutrient) => viewNutrients.includes(nutrient.name))
    : [];

  function getTitle(name: string) {
    switch (name) {
      case 'Calories':
        return 'kcal';
      case 'Fat':
        return 'Fat';
      case 'Carbohydrates':
        return 'Carbs';
      case 'Protein':
        return 'Protein';
      default:
        return name;
    }
  }

  return (
    <Box sx={{ display: 'flex', flexDirection: 'row', gap: '0.5rem', mb: 1 }}>
      <Typography variant="body1_Bold" color={'grey.dark'}>
        {title}
      </Typography>
      {filteredNutrients.map(({ name, amount, unit }) => (
        <Box
          key={name}
          sx={{
            display: 'flex',
            flexDirection: 'column',
            padding: '0.5rem',
            backgroundColor: 'success.light',
            borderRadius: '0.5rem',
            alignItems: 'center',
          }}
        >
          <Typography
            variant="body1_Bold"
            color={'grey.dark'}
            sx={{ mb: 0 }}
            fontSize="small"
          >
            {getTitle(name)}
          </Typography>
          <Typography
            variant="body1"
            color={'grey.dark'}
            sx={{ mb: 0 }}
            fontSize="small"
          >
            {amount} {name === 'Calories' ? '' : unit}
          </Typography>
        </Box>
      ))}
    </Box>
  );
}

export default Nutrients;
