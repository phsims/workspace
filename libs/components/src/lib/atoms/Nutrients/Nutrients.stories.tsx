import type { Meta, StoryObj } from '@storybook/react';
import { Nutrients } from './Nutrients';
import { mockNutrients } from '../../__mocks__/nutrientsData';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof Nutrients> = {
  component: Nutrients,
  title: '@atoms/Nutrients',
};
export default meta;
type Story = StoryObj<typeof Nutrients>;

export const Primary = {
  args: {
    nutrients:  mockNutrients,
  },
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to Nutrients!/gi)).toBeInTheDocument();
  },
};
