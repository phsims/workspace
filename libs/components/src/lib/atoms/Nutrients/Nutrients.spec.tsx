import { render } from '@testing-library/react';
import { mockNutrients } from '../../__mocks__/nutrientsData';
import Nutrients from './Nutrients';

describe('Nutrients', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Nutrients nutrients={mockNutrients} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<Nutrients nutrients={mockNutrients} />);
    expect(baseElement).toMatchSnapshot();
  });
});
