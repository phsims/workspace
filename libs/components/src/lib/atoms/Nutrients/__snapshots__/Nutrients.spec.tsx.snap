// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nutrients should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1hf8v5l"
    >
      <span
        class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
      />
      <div
        class="MuiBox-root css-c2tbbu"
      >
        <span
          class="MuiTypography-root MuiTypography-body1_Bold css-1el569k-MuiTypography-root"
        >
          kcal
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1k4wxgx-MuiTypography-root"
        >
          316.49
           
        </p>
      </div>
      <div
        class="MuiBox-root css-c2tbbu"
      >
        <span
          class="MuiTypography-root MuiTypography-body1_Bold css-1el569k-MuiTypography-root"
        >
          Fat
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1k4wxgx-MuiTypography-root"
        >
          12.09
           
          g
        </p>
      </div>
    </div>
  </div>
</body>
`;
